### PR TITLE
portal: Add ability to modify/remove by index

### PR DESCRIPTION
**Pull request by sophie-h**
This feature is handy for key managers that edit entries

### CODE REVIEW

**Comment by sophie-h** [comment]
Still have to test the delete function with Key Rack

**Comment by bilelmoussaoui** [comment]
Could use a rebase

**Comment by sophie-h** [comment]
> Could use a rebase

I mean yeah, it's based on the mentioned PR and rebases just fine on main now

File: src/portal/error.rs [hunk 1]
```diff
@@ -79,7 +82,37 @@ impl std::fmt::Display for Error {
...
+#[derive(Debug)]
+/// All information that is available about an invalid (not decryptable)
+/// [`Item`](super::Item)
+pub struct InvalidItemError {
+    error: Error,
+    attribute_names: Vec<String>,
+}
+
+impl InvalidItemError {
+    pub fn new(error: Error, attribute_names: Vec<String>) -> Self {
```
Comment (bilelmoussaoui) [inline src/portal/error.rs:100]: Should the constructor be public?

File: src/portal/mod.rs [hunk 2]
```diff
@@ -195,6 +206,44 @@ impl Keyring {
         self.write().await
     }
 
+    /// Replaces item at the given index.
+    ///
+    /// The `index` refers to the index of the [`Vec`] returned by
+    /// [`items()`](Self::items). If the index does not exist, the functions
+    /// returns an error.
+    pub async fn replace_item_index(&self, index: usize, item: &Item) -> Result<(), Error> {
```
Comment (bilelmoussaoui) [inline src/portal/mod.rs:214]: Not a fan of the name of those two methods but I don't have a better suggestion neither

Comment (sophie-h) [reply src/portal/mod.rs:214]: Me neither

**Review by bilelmoussaoui — APPROVED** [review]

**Pull request merged**